src/sentry/static/sentry/app/utils/discover/discoverQuery.tsx

```diff
@@ -197,20 +198,24 @@ class DiscoverQuery extends React.Component<Props, State> {
     apiPayload: EventQuery & LocationQuery & TrendsQuery,
     location: Location
   ) => {
-    const {trendChangeType, eventView} = this.props;
+    const {trendChangeType, eventView, isTrendsProject} = this.props;
     if (trendChangeType) {
       delete apiPayload.cursor;
-      if (
-        trendChangeType === TrendChangeType.IMPROVED &&
-        location?.query?.improvedCursor
-      ) {
-        apiPayload.cursor = location?.query?.improvedCursor;
-      } else if (
-        trendChangeType === TrendChangeType.REGRESSION &&
-        location?.query?.regressionCursor
-      ) {
-        apiPayload.cursor = location?.query?.regressionCursor;
+
+      if (!isTrendsProject) {
+        if (
+          trendChangeType === TrendChangeType.IMPROVED &&
+          location.query.improvedCursor
+        ) {
+          apiPayload.cursor = location.query.improvedCursor;
+        } else if (
+          trendChangeType === TrendChangeType.REGRESSION &&
+          location.query.regressionCursor
+        ) {
+          apiPayload.cursor = location.query.regressionCursor;
```

Comment:
I think I've conflicted you on this hunk. The upside of that is that you can now pass a cursor prop to DiscoverQuery which could help save you some logic.

Reply:
Yep I kind of expected you would with that other pr. I guess I can just pass cursor in now and it will hopefully work 👍